File: python-algorithm/algo_03_two_points/L633.cpp
//
// Created by LiHeng on 2020/6/3.
//
#include <iostream>
#include <cmath>
using namespace std;
class Solution {
public:
    bool judgeSquareSum(int c) {
        int i = 0;
        int j = (int)sqrt(c);
        while(i<=j){
            int sum = -c + i*i + j*j; //一开始使用的是i*i+j*j 太大的数字会报错，然后用i*i+j*j-c还是报错，最后用-c+i*i+j*j
            if(sum == 0)
                return true;
            else if (sum < 0)
                ++i;
            else
                --j;
        }
        return false;
    }
};

File: python-algorithm/algo_03_two_points/L633.py
#--coding:utf-8--
'''
@Time   : 2020/9/24
@Author : Heng Li
@Email  : liheng_lulia@163.com
'''

# 题目只说是2个整数，没说不能是2个一样的整数
import math
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        lt = 0
        rt = int(math.sqrt(c))
        while lt <= rt:
            s = lt*lt +rt*rt
            if s > c:
                rt -= 1
            elif s < c:
                lt += 1
            else:
                return True
        return False